### PR TITLE
Refactors web.AuthManager Constructor to use the options pattern

### PR DESCRIPTION
**Pull request by wcharczuk**
## PR Summary

Refactors web.AuthManager Constructor to use the options pattern.

As a result, the top level `web.New()` constructor will now also return an error. A `MustNew()` form has been added for convenience and to smooth refactoring.

It also adds `.Header` to the response logger event metadata for use in tactical situations where you want to test for things in the headers. 

Some other smaller tweaks:
- removes the logger reference from `web.Ctx`
- changes slightly a few function names on logger metadata maps
- addresses inconsistencies with web.ResultProvider implementations returning 401 vs. 403 (initiated previous to this PR).

 - **Type:** Internal
 - **Intended Change Level:** major

#### Reviewers:

Reviewers keep the following in mind while reviewing this PR, and provide an assessment of them in your approval comment:

- Does the "Intended Change Level" above match the actual behavior of this PR?
- Is this PR following patterns set forth in the package (if modifying a package), or the go-sdk design patterns if implementing a new package from scratch?
- Does this require a backport to LTS versions? If so ping, let the contributors group know.

### CODE REVIEW

File: web/auth_manager.go [hunk 1]
```diff
@@ -9,76 +9,189 @@ import (
...
-	manager.CookiePath = cfg.CookiePathOrDefault()
-	manager.CookieSameSite = cfg.CookieSameSiteOrDefault()
-	manager.SessionTimeoutProvider = SessionTimeoutProvider(!cfg.SessionTimeoutIsRelative, cfg.SessionTimeoutOrDefault())
-	return manager
+func NewAuthManager(options ...AuthManagerOption) (manager AuthManager, err error) {
+	manager.CookieDefaults.Name = DefaultCookieName
+	manager.CookieDefaults.Secure = DefaultCookieSecure
+	manager.CookieDefaults.HttpOnly = DefaultCookieHTTPOnly
+	manager.CookieDefaults.SameSite = DefaultCookieSameSite
+
```
Comment (alee101) [inline web/auth_manager.go]: should Path also be defaulted here?

File: web/ctx.go [hunk 2]
```diff
@@ -30,22 +29,47 @@ func NewCtx(w ResponseWriter, r *http.Request, options ...CtxOption) *Ctx {
...
-	Views           *ViewCache
-	Log             logger.Log
-	Tracer          Tracer
-	Body            []byte
-	Form            url.Values
+	// App is a reference back to the parent application.
+	App *App
+	// Auth is a reference to the app default auth manager, but
+	// can be overwritten by middleware.
+	Auth AuthManager
```
Comment (alee101) [inline web/ctx.go]: for what cases should the context level `ctx.Auth` be used over `ctx.App.Auth`?

File: web/auth_manager.go (hunk 1 above)
Comment (wcharczuk) [reply web/auth_manager.go]: yep, 100%

File: web/ctx.go (hunk 2 above)
Comment (wcharczuk) [reply web/ctx.go]: Thinking is you can override the app default with middleware on a per route / controller basis if you want to switch how different "areas" of an app authenticate.

File: web/auth_manager.go (hunk 1 above)
Comment (wcharczuk) [reply web/auth_manager.go]: done

**Pull request merged**